material_studies/rally/lib/finance.dart

```diff
@@ -0,0 +1,333 @@
+}
+
+List<FinancialEntityCategoryView> fromBillDataList(List<BillData> items) {
+  return List<FinancialEntityCategoryView>.generate(
+      items.length, (i) => fromBillItem(items[i], i));
+}
+
+List<FinancialEntityCategoryView> fromBudgetData(
+    List<BudgetData> items, BuildContext context) {
+  return [
+    for (var i = 0; i < items.length; i++) fromBudgetItem(items[i], i, context)
+  ];
+//    List<FinancialEntityCategoryView>.generate(
+//      items.length, (i) => fromBudgetItem(items[i], i, context));
+}
+
+class FinancialEntityCategoryDetailsPage extends StatelessWidget {
+  final List<DetailedEventData> items =
+      DummyDataService.getDetailedEventItems();
+
+  @override
+  Widget build(BuildContext context) {
+    final List<_DetailedEventCard> cards = items
+        .map((i) => _DetailedEventCard(
+              title: i.title,
+              subtitle: Formatters.date.format(i.date),
+              amount: i.amount,
+            ))
+        .toList();
+
+    return Scaffold(
+      appBar: AppBar(
+        elevation: 0.0,
+        centerTitle: true,
+        title: Text(
+          'Checking',
+          style: Theme.of(context).textTheme.body1.copyWith(fontSize: 18.0),
+        ),
+      ),
+      body: Column(children: [
+        SizedBox(
+            height: 200.0,
+            width: double.infinity,
+            child: RallyLineChart(events: items)),
+        Flexible(
+          child: ListView(shrinkWrap: true, children: cards),
+        )
+      ]),
+    );
+  }
+}
+
+class _DetailedEventCard extends StatelessWidget {
+  const _DetailedEventCard({
+    @required this.title,
+    @required this.subtitle,
+    @required this.amount,
+  });
+
+  final String title;
+  final String subtitle;
+  final double amount;
+
+  @override
+  Widget build(BuildContext context) {
+    return FlatButton(
+      onPressed: () {},
+      child: SizedBox(
+        height: 68.0,
+        child: Column(
+          children: [
+            SizedBox(
+              height: 67.0,
+              child: Row(
+                children: [
+                  Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: [
+                      Text(
+                        title,
+                        style: Theme.of(context)
+                            .textTheme
+                            .body1
+                            .copyWith(fontSize: 16.0),
+                      ),
+                      Text(
+                        subtitle,
+                        style: Theme.of(context)
+                            .textTheme
+                            .body1
+                            .copyWith(color: RallyColors.gray60),
+                      )
+                    ],
+                  ),
+                  Spacer(),
+                  Text(
+                    '\$ ' + Formatters.usd.format(amount),
```

Comment:
String interpolation would be more idiomatic here:

`'\$${Formatters.usd.format(amount)}'`

Reply:
Done